internal/locate/region_cache.go

```diff
@@ -1123,9 +1093,18 @@ func (c *RegionCache) findRegionByKey(bo *retry.Backoffer, key []byte, isEndKey
 			c.insertRegionToCache(r, true, true)
 			c.mu.Unlock()
 		}
-	} else if r.checkNeedReloadAndMarkUpdated() {
+	} else if flags := r.resetSyncFlags(needReloadOnAccess | needAsyncReloadReady); flags > 0 {
```

Comment:
Seems the async reload becomes sync here.

Reply:
Rename or comment the flag, both OK to me.

The current comment indicates it will be reloaded in async mode, which is inaccurate.

```go
	needAsyncReloadPending                   // indicates the region will be reloaded in async mode
	needAsyncReloadReady                     // indicates the region is ready to be reloaded in async mode
```

Reply:
Renamed, PTAL.